File: CSocketComm/cmd.c
#include "cmd.h"
#include<stdio.h>
#include<string.h>

char* UNKNOWN = "unknown";
char* ERROR = "error";
char* STOPPED = "stopped";
char* EXIT =  "exit";
char* GO = "go";
char* DONE = "done";
char* READY = "ready";
char* OK = "ok";

CMD parseCmd(char* cmd){
    printf("[%s]\n", cmd);
    if(0==strcmp(ERROR, cmd)) return Error;
    if(0==strcmp(STOPPED, cmd)) return Stopped;
    if(0==strcmp(EXIT, cmd)) return Exit;
    if(0==strcmp(GO, cmd)) return Go;
    if(0==strcmp(DONE, cmd)) return Done;
    if(0==strcmp(OK, cmd)) return Ok;

    // if(0==strcmp(Unknown, cmd)) 
        // return Unknown;
    return Unkown;
    
}
char* tos(CMD cmd){
    switch (cmd)
    {
        case Error:   return ERROR;
        case Stopped: return STOPPED;
        case Exit:    return EXIT;
        case Go:      return GO;
        case Done:    return DONE;
        case Ok:      return OK;
        case Ready:   return READY;
        default:      return UNKNOWN;
    }
}

File: JSocketComm/src/JSocketComm/Cmd.java
package JSocketComm;

import java.util.EnumSet;
import java.util.HashMap;
import java.util.Map;

public enum Cmd {
    Unknown((byte)-3),
    Error((byte)-2),
    Stopped((byte)-1),
    Exit((byte)0),
    Go((byte)1),
    Done((byte)2),
    Ready((byte)3),
    Ok((byte)4),
    Receive((byte)5);
    private static final Map<Byte, Cmd> VALUE_TO_ENUM_MAP;
    private final byte value;

    static {
        VALUE_TO_ENUM_MAP = new HashMap<>();
        for (Cmd type : EnumSet.allOf(Cmd.class)) {
            VALUE_TO_ENUM_MAP.put(type.value, type);
        }
    }

    private Cmd(byte value) {
        this.value = value;
    }

    public byte getValue() {
        return value;
    }
    public static Cmd forValue(byte value) {
        return VALUE_TO_ENUM_MAP.get(value);
    }
}